rasterio/_io.pyx

```diff
@@ -482,7 +485,9 @@ cdef class DatasetReaderBase(DatasetBase):
                 out.shape = (1,) + out.shape
 
         if not indexes:
-            raise ValueError("No indexes to read")
+            out = np.array()
+            logging.warning("No indexes to read, return empty array")
+            return out
```

Comment:
I'm not sure about this, empty indexes feels just wrong to me and should raise an issue (as it is doing right now). I'm afraid that empty arrays could lead to misunderstandings of the read behaviour.

Reply:
Ok got it 
But the idea is to find a way to compute 
the `.meta` or even the `.bounds`  of a `DatasetReader` without loading the entire raster data 

I don't know if current version of rasterio would be able to do this ?

Reply:
you don't need to `read` the data to get the `meta` nor the `bounds` but just `open` the data

```python
with rasterio.open("file.tif") as src:
    print(src.meta)
    print(src.bounds)
```

Reply:
yes, 
For sure 

But 

lets imagine i want to merge two rasters  (from 2 distinct `tiff` files)
```
merged_img, merged_transform = merge(['a.tif', 'b.tif'], indexes=1)
```
In this case, 
We need to load at least 1 band in order to have the `merged_transform`

Reply:
Are you wanting to create a template raster without the data being transferred from the rasters?

Reply:
Yes for example

Reply:
What you probably want is: https://rasterio.readthedocs.io/en/latest/api/rasterio.transform.html#rasterio.transform.from_bounds

Then, instead of using `merge` or `crop`, you can get the bounds of the raster(s) and polygons handle them that way. Shapely would likely be a useful tool for that: https://shapely.readthedocs.io/en/stable/manual.html

You may also be interested in the GeoBox concept provided by odc-geo: https://odc-geo.readthedocs.io/en/latest/intro-geobox.html

Reply:
ok then it's not an issue with `read()`. You can simply write something like 
```python
def get_transform(datasets: Sequence[str], target_aligned_pixels: bool = False):

    # scan input files
    xs = []
    ys = []
    resolutions = []
    for dataset in datasets:
        with rasterio.open(dataset) as src:
            left, bottom, right, top = src.bounds
            resolutions.extend(src.res)
        xs.extend([left, right])
        ys.extend([bottom, top])

    dst_w, dst_s, dst_e, dst_n = min(xs), min(ys), max(xs), max(ys)

    # Select the first resolution (this could be configurable)
    res = resolutions[0]


    if target_aligned_pixels:
        dst_w = math.floor(dst_w / res[0]) * res[0]
        dst_e = math.ceil(dst_e / res[0]) * res[0]
        dst_s = math.floor(dst_s / res[1]) * res[1]
        dst_n = math.ceil(dst_n / res[1]) * res[1]

    # Compute output array shape. We guarantee it will cover the output
    # bounds completely
    output_width = int(round((dst_e - dst_w) / res[0]))
    output_height = int(round((dst_n - dst_s) / res[1]))

    output_transform = Affine.translation(dst_w, dst_n) * Affine.scale(res[0], -res[1])
    return output_transform
```

Reply:
Ook thank you 🙏 
Do you think it would be a nice future feature for Rasterio ?